### PR TITLE
feat(apim): facility amendment

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction
APIM TFS migration

## Resolution
* Added `facilityIdentifier` to facility loan and fixed fee durable activity function, invoked upon facility issuance.
* Extracted facility loan and fixed fee ACBS response objects for ingestion into `tfm.acbs` object.

## Miscellaneous
* Number generator micro-service improved logging.
* Number generator code refactorisation.
* Introduced `dealHasAllValidUkefIds` to mitigate https://ukef-dtfs.atlassian.net/browse/DTFS2-5991.

### CODE REVIEW

**Review by IgnatG — APPROVED** [review]

File: trade-finance-manager-api/src/v1/helpers/dealHasAllUkefIds.js [hunk 1]
```diff
@@ -11,15 +13,41 @@ const dealHasAllUkefIds = async (dealId) => {
...
+/**
+ * Verify whether the deal and all its facilities have a valid UKEF ID.
+ * 10 digit numerical ID.
+ * @param {String} dealId UKEF tfmDeal ID
+ * @returns {Boolean}  Boolean value
+ */
+const dealHasAllValidUkefIds = async (dealId) => {
+  const tfmDeal = await findOneTfmDeal(dealId);
+
+  if (tfmDeal && tfmDeal.dealSnapshot && tfmDeal.dealSnapshot.facilities) {
```
Comment (IgnatG) [inline trade-finance-manager-api/src/v1/helpers/dealHasAllUkefIds.js:36]: we can simplify this into
```js
if (tfmDeal?.dealSnapshot?.facilities) {.. }
```

Comment (IgnatG) [reply trade-finance-manager-api/src/v1/helpers/dealHasAllUkefIds.js:36]: not a show stopper though

Comment (abhi-markan) [reply trade-finance-manager-api/src/v1/helpers/dealHasAllUkefIds.js:36]: Good spot, have added.

**Review by IgnatG — APPROVED** [review]
LGTM

**Pull request merged**